geomstats/geometry/connection.py

```diff
@@ -125,6 +125,21 @@ def exp(
         exp : array-like, shape=[..., dim]
             Point on the manifold.
         """
+        base_point = gs.broadcast_to(base_point, tangent_vec.shape)
+        n_base_point, n_tangent_vec = base_point.shape[0], tangent_vec.shape[0]
```

Comment:
I think we can now remove the code from this line till l142: this portion of code does not really do anything and the line 129 does not work properly for the case 1 base point, 1 tangent vec